testament/testament.nim

```diff
@@ -79,6 +83,20 @@ type
     startTime: float
     debugInfo: string
 
+  CompilerOutput = object
+    ## Describes the output of a a compiler invocation.
+    cmd: string    ## the command that was run
+    nimout: string ## compiler output
+    output: string ## test output
+
+    # information related to reported compiler errors (unused if
+    # none were reported):
+    file, msg: string
+    line, column: int
```

Comment:
And I think I would've used a dedicated type for the <file:line:col> triple (if we don't have suitable, then create) and put the whole thing as `Optional[]` so it would've be two explicitly optional fields. 


Or use 4-tuple of file:line:col:msg with the same wrapper.

Reply:
But that's just IMO, so can be ignored

Reply:
Yep, I agree that bundling them makes sense, but I wanted to keep the amount of changes small, so decided against it.